File: src/test/java/pages/LoginPage.java
package pages;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;

public class LoginPage extends BasePage {

    private final Locator userEmailField = placeholder("username@gmail.com");
    private final Locator userPasswordField = placeholder("Your password");
    private final Locator loginButton = button("Login");

    public LoginPage(Page page) {
        super(page);
    }

    public LoginPage fillUserEmailField(String userEmail) {
        userEmailField.fill(userEmail);
        return new LoginPage(getPage());
    }

    public LoginPage fillUserPasswordField(String userPassword) {
        userPasswordField.fill(userPassword);
        return new LoginPage(getPage());
    }

    public HomePage clickLoginButton() {
        loginButton.click();
        return new HomePage(getPage());
    }
}

File: page_objects/loginPage.js
import AbstractBaseSet from "./abstractBaseSet";
import SignPage from "./signPage";

class LoginPage extends AbstractBaseSet {
    constructor(page) {
        super(page);
        // this.page = page;
    // }

    this.locators = {
        ...this.locators,
        getEmailAddressInputField: () => this.page.getByPlaceholder('username@gmail.com'), 
        getPasswordInputField: () => this.page.getByPlaceholder('Your password'),        
        getLoginBtn: () => this.page.getByRole('button', {name: 'Login'}),
    }
}

    async fillEmailAddressInputField(email){
        await this.locators.getEmailAddressInputField().fill(email);

        return this;
    }

    async fillPasswordInputField(password) {
        await this.locators.getPasswordInputField().fill(password);

        return this;
    }

    async clickLoginAndGoSignPage() {
        await this.locators.getLoginBtn().click();

        return new SignPage(this.page);
    }
}
export default LoginPage;